hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -712,9 +776,14 @@ private AbfsRestOperation conditionalCreateOverwriteFile(final String relativePa
             .getResponseHeader(HttpHeaderConfigurations.ETAG);
 
         try {
-          // overwrite only if eTag matches with the file properties fetched befpre
-          op = client.createPath(relativePath, true, true, permission, umask,
-              isAppendBlob, eTag, tracingContext);
+          // overwrite only if eTag matches with the file properties fetched before.
+          if (getPrefixMode() == PrefixMode.BLOB) {
+            op = client.createPathBlob(relativePath, true, true, metadata,
+                    eTag, tracingContext);
+          } else {
+            op = client.createPath(relativePath, true, true, permission, umask,
+                    isAppendBlob, eTag, tracingContext);
+          }
```

Comment:
There are 3 places that need to createPath. abstracting creation (line 780 to 786) and returning op will reduce code duplication.

Reply:
done